### PR TITLE
Migrate CSAM reporting data access logic to pg-promise

### PR DESCRIPTION
**Pull request by GPaoloni**
Primary reviewer: @stephenhand 

## Description
This PR:
- Moves the current CSAM endpoints data access logic to be implemented using pg-promise / Typescript rather than Sequelize / JS.
- Moves the code into the new HRM module layout (grouped by function).
- When a contact is created, the step of connecting the csam reports to the newly created contact is a separate step in the data access layer, instead than all being the same query. I can rollback this change but IMO being more explicit about a different table being modified in this operation makes worth the trade-off of losing efficiency in the query.

I've also done an exhaustive read on all the places where csam reports are being queried alongside the contacts, and they are queried by the FK `contactId-accountSid` in every occurrence (I don't know what I saw when raised that concern o.O Sorry!) 

After this PR, only post survey components are referencing the Sequlize based `models` :D
Screenshot from 2022-11-17 16-43-36

### Checklist
- [x] [Corresponding issue has been opened](https://bugs.benetech.org/browse/CHI-1485)
- [x] New tests added

### Verification steps
- Start local server from `master` branch, open a Flex instance, save some csams and a new contact so they are linked.
- Checkout this branch and restart the server.
- Confirm that retrieving the contact saved with the previous version still works as intended (the csam reports are bundled with the contact).
- Save another contact with csam reports and confirm that they are properly linked.
- Regression test to make sure I'm not breaking stuff.
- Make a POST to `/csamReports` endpoint, providing a valid `contactId` in the payload, and confirm that this newly created report is linked to the corresponding contact.

### CODE REVIEW

**Comment by stephenhand** [comment]
Looks good. 

I think we have an opportunity to remove a bunch of models now we only have postSurvey in Sequelize? Since the only DB relationship we care about Sequelize enforcing is the PostSurvey<>Contact, could we remove the other relationships from the contact model? Then all the other models except Contact and PostSurvey could be deleted, and that last controller - `CaseAuditController`.

**Review by stephenhand — APPROVED** [review]

**Comment by stephenhand** [comment]
Also, should we do a migration to delete any CSAM reports arrays from the JSON blobs if we're confident they haven't been used for a while, just to tidy up?

File: hrm-service/src/csam-report/csam-report-routes-v0.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { SafeRouter, publicEndpoint } from '../permissions';
```
[the remark this answers is not in the document]
Comment (GPaoloni) [reply hrm-service/src/csam-report/csam-report-routes-v0.ts:1]: This one is my fault, copy pasted instead than `mv`ing around.

**Comment by GPaoloni** [comment]
>Also, should we do a migration to delete any CSAM reports arrays from the JSON blobs if we're confident they haven't been used for a while, just to tidy up?

@stephenhand After reviewing the code, since the [new "contact create" has been introduced](https://github.com/techmatters/hrm/commit/752095005a02a3bf0f762a11604802a4cfc5a633), we are not saving the csam list as part of the contact's json, neither we did before as can be seen in the [PR the old controller was removed](https://github.com/techmatters/hrm/commit/f00cbbbccf427812d0b14869c398fd0f2d01f41e). I can run some queries to confirm this if you think is a good idea tho.

I remember seeing a different thing when we chatted about this, but it's likely that I got confused at that time. What do you think?

**Pull request merged**